ceph/rbd/utils.py

```diff
@@ -0,0 +1,180 @@
+import random
+import string
+
+from ceph.ceph import CommandFailed
+from utility.log import Log
+
+log = Log(__name__)
+
+
+getdict = lambda x: {k: v for (k, v) in x.items() if isinstance(v, dict)}
+
+isdict = lambda x: [isinstance(i, dict) for i in x]
+
+
+def find(key, dictionary):
+    """Find list of values for a particular key from a nested dictionary."""
+    for k, v in dictionary.items():
+        if k == key:
+            yield v
+        elif isinstance(v, dict):
+            for result in find(key, v):
+                yield result
+        elif isinstance(v, list):
+            for d in v:
+                if isinstance(d, dict):
+                    for result in find(key, d):
+                        yield result
+                else:
+                    yield d
+
+
+def value(key, dictionary):
+    """Retrieve first value of a particular key from a nested dictionary."""
+    return str(list(find(key, dictionary))[0])
+
+
+def copy_file(file_name, src, dest):
+    contents, err = src.exec_command(sudo=True, cmd="cat {}".format(file_name))
+    key_file = dest.remote_file(sudo=True, file_name=file_name, file_mode="w")
+    key_file.write(contents)
+    key_file.flush()
+
+
+def get_md5_sum(**kw):
+    """
+    kw: {
+        "image_spec": <>,
+        "file_path": <>,
+        "rbd": <rbd_object>,
+        "client": <client_node>
+    }
+    """
+    if kw.get("image_spec"):
+        export_spec = {
+            "source-image-or-snap-spec": kw.get("image_spec"),
+            "path-name": kw.get("file_path"),
+        }
+        if kw.get("rbd").export(**export_spec):
+            log.error(f"Export failed for image {kw.get('image_spec')}")
+            return None
+    return exec_cmd(
+        output=True, cmd="md5sum {}".format(kw.get("file_path")), node=kw.get("client")
+    ).split()[0]
+
+
+def check_data_integrity(**kw):
```

Comment:
The idea is to completely deprecate the existing RBD and RBD Mirror classes once all our tests have been adopted to SDK. So I will not be using any existing methods from those classes, and I will be creating duplicate methods for them here in the new utils.

Also since check_data_integrity is not coupled to an RBD or Mirror class, it need not be part of that class, it could be in a common utils file.